be/src/exec/tablet_sink.cpp

```diff
@@ -168,6 +168,16 @@ Status OlapTableSink::init(const TDataSink& t_sink, RuntimeState* state) {
         _colocate_mv_index = table_sink.enable_colocate_mv_index && config::enable_load_colocate_mv;
     }
 
+    // Query context is only available for pipeline engine
+    auto query_ctx = state->query_ctx();
+    if (query_ctx) {
```

Comment:
using DCHECK(query_ctx) is just ok, if query_ctx is nullptr, the catastrophe must happens, the BE should not go on.

Reply:
For stream load/routine load, there is no query_ctx because they still use non-pipeline engine. The load channel profile is only available for broker load/insert, so need to check the query_ctx.